### ISSUE TITLE
Problem with connect 3.3.5 (Missing required parameter: scope)

### ISSUE THREAD

**Issue by hugovart** [jaredhanson/passport-google-oauth#80]
I had a working app with `passport-google-oauth v0.1.5` and `passport v0.1.16` on `node v0.8` with `connect v2.27.1`. The issue appeared since I updated to `connect v3.3.5` on `node v0.10.38` and  `passport-google-oauth v0.2.0` and `passport v0.2.1`.

The first request looks OK to me.
```
https://accounts.google.com/o/oauth2/auth?access_type=offline&approval_prompt=force&user_id=testaccount%40gmail.com&response_type=code&redirect_uri=http%3A%2F%2Fapp.testapp.com%3A5000%2Foauth2%2Fcallback&scope=profile%20email&client_id=XXXXXXXX.apps.googleusercontent.com
```
After click the `Accept` button at the permission page I see this problem.
```
400. That’s an error.
Error: invalid_request
Missing required parameter: scope
Request Details
response_type=code
redirect_uri=http://app.testapp.com:5000/oauth2/callback
client_id=XXXXXXXX.apps.googleusercontent.com
That’s all we know.
```

Here are the details, how I set up.
```
passport.use new GoogleStrategy(
  clientID: process.env.GOOGLE_CLIENT_ID || conf.GOOGLE_CLIENT_ID
  clientSecret: process.env.GOOGLE_CLIENT_SECRET || conf.GOOGLE_CLIENT_SECRET
  callbackURL: "https://" + process.env.HOST) + "/oauth2/callback"
  passReqToCallback: true
, (req, accessToken, refreshToken, profile, done) ->
```
```
server = connect()
  .use(cookieParser())
  .use(bodyParser.json())
  .use(bodyParser.urlencoded({extended: true}))
  .use(session(
    store: new RedisStore(
       pass: getRedisParams().pass
       host: getRedisParams().host
       port: getRedisParams().port
       ttl: 3600 #one hour
     )
    secret: "XXXXXXXX"
     saveUninitialized: true
     resave: true
     cookie:
       domain: process.env.HOST || conf.HOST
       maxAge: 604800000
  ))
  .use(passport.initialize())
  .use(passport.session())
  .use(connectRoute(routes))
```
```
(passport.authenticate('google',
    scope: [
      'profile'
      'email'
    ]
    accessType: 'offline'
    userID: email
    approvalPrompt: 'force'
  )) req, res, next
```
```
    (passport.authenticate('google',
      failureRedirect: '/oauth2/failed'
      successRedirect: '/oauth2/success'
    )) req, res, next
```

**Comment by celalo** [comment]
I have the same problem. I could not use passport-google-oauth with connect 3.x and node 0.10.x

**Comment by hugovart** [comment]
I am stuck at this with for over a week now. It blocks me upgrading to node v.10 because node v0.10 needs connect v3 and connect v3 does not work with the latest `passport-google-oauth`. I've been trying to find a solution but I still don't have a clue. I tried with playing with sessions, changed session store, tried various different options `saveUninitialized:true/false`, `bodyParser.urlencoded({extended: true/false}`, `resave: true/false` etc. It didn't help. Made trials with different scopes etc.
Could anyone at least tell me, if anyone is successful at using this module with connect 3x or express 4x?

**Comment by LJWall** [comment]
The example (updated in pull request https://github.com/jaredhanson/passport-google-oauth/pull/81) works for me on node v0.12.0 and express 4.12.13

**Comment by hugovart** [comment]
Yes, I managed to have it worked with v4.12.3 on node v0.10.38 following the instructions in pull request #81. Still could not figure out why it does not work with connect v3.3.5.

**Comment by jbcpollak** [comment]
I seem to be having this same problem with connect v3.4.0. I'm trying to avoid switching to express as I'm not the lead on this project.

**Comment by keith24** [comment]
Just in case this helps anyone coming from google, I was having this error when doing: 
```javascript
app.get('/login/google/callback', (req,res) => {
  passport.authenticate( 'google', {failureFlash:true} );
  // Callback
});
```
I guess the call to `passport.authenticate()` has to be directly in the express route's callback chain: 
```javascript
app.get('/login/google/callback', passport.authenticate( 'google', {failureFlash:true} ),
  (req,res) => {
    // Callback
  }
);
```
That's working for me now.

**Comment by Ekzer** [comment]
For anyone who gets the scope problem, pass it to Passport params like this: (it works perfect for me)

`new GoogleStrategy({
            clientID: global.config.passport.google.clientID,
            clientSecret: global.config.passport.google.clientSecret,
            callbackURL:  "http://localhost:3000/login/google/callback",//global.config.host+global.config.passport.google.callbackUrl,
            usernameField : "email",
            passwordField : "password",
            scope: "https://www.googleapis.com/auth/plus.login",
            passReqToCallback : true // allows us to pass back the entire request to the callback
            },
            function (req, token, refreshToken, profile, done) {
                token;
            })`
